### DISCUSSION TITLE
feat: Chat profile

### DISCUSSION THREAD

**Discussion by WilsonZiweiWang** [open-webui/open-webui#4394] [Ideas]
**Is your feature request related to a problem? Please describe.**
When users start a new chat, several steps could be involved. First, they select an LLM, then they load the documents for RAG. Finally,  they might want to change the system prompt and configure the advanced parameters on LLM and RAG for optimal results. That is ok for the first time, but if other people ever need to use the same setup, they will have to go through the whole process again.

**Describe the solution you'd like**
The idea I have in mind is similar to the GPTs by OpenAI. For webui, we could let the admin have the option to add chat profiles in which they can configure the LLM and embedding model to use, documents to be embedded as well as other parameters such as system prompt and chunk size. Once the profile is created, users can select the profile when they try to start a new chat. The new chat will use everything that is pre-defined in the profile. The admin might also restrict access to certain profiles for some users if they are not allowed to access the documents specified in the profile.

**Comment by gomesdigital** [comment]
We want this feature so that an admin can filter content for _all_ users, e.g. a global system prompt. It might say "Do not offer or answer any questions on financial advice."

The "profiles" suggestion could solve this if we're able to limit access to them.

**Comment by WilsonZiweiWang** [comment]
Some proof of concept for the chat profile with knowledge bases:

## Chat profile
![image](https://github.com/user-attachments/assets/5ea433c6-ad5a-4f0f-ba35-a4a7ece3a204)
![image](https://github.com/user-attachments/assets/3cc23d0f-6882-48da-8bbf-b48bb25c24d1)
![image](https://github.com/user-attachments/assets/5f8ccd96-bb21-4e4b-a4cc-ec4009fce8bb)
![image](https://github.com/user-attachments/assets/fcc5856b-24a9-4e7f-b7f2-e2add0111ec6)

---

## Knowledge base
![image](https://github.com/user-attachments/assets/5b570559-736e-4303-ac9b-64985cf41fc8)
![image](https://github.com/user-attachments/assets/3a302b27-d493-4c88-93a9-71933e5ecd54)
![image](https://github.com/user-attachments/assets/7e258f31-b110-4e14-b49d-f50f03154f17)
![image](https://github.com/user-attachments/assets/b0663de6-2f47-417f-8ed6-0a53a5a0662b)

---

## Create a chat from a profile
[Screencast from 2024-10-16 16-46-04.webm](https://github.com/user-attachments/assets/849ddb08-639d-4792-b73a-a14b25685e27)

---
To try out: https://github.com/WilsonZiweiWang/open-webui/tree/chat-profiles-and-knowledge-bases
